classes/Kohana/Log/Writer.php

```diff
@@ -67,24 +62,202 @@
 	}
 
 	/**
+	 * Gets the date/time format for writing a timestamp
+	 * 
+	 * @return string
+	 */
+	public function get_timestamp_format()
+	{
+		return $this->timestamp_format;
+	}
+
+	/**
+	 * Sets the date/time format for writing a timestamp
+	 * 
+	 * @param string $timestamp_format
+	 * @return Log_Writer
+	 */
+	public function set_timestamp_format($timestamp_format)
+	{
+		$this->timestamp_format = (string) $timestamp_format;
+		
+		return $this;
+	}
+	
+	/**
+	 * Gets the log level used when tracing an exception stack
+	 * 
+	 * @return string the PSR-3 log level used for stack tracing
+	 */
+	public function get_strace_level()
+	{
+		return $this->strace_level;
+	}
+
+	/**
+	 * Sets the log level used when tracing an exception stack
+	 * 
+	 * @param mixed $level
+	 * @throws InvalidArgumentException
+	 * @return Log_Writer
+	 */
+	public function set_strace_level($level)
+	{
+		$this->strace_level = Log::to_psr_level($level);
+		
+		return $this;
+	}
+	
+	/**
+	 * Gets the default string format of the log entry
+	 * 
+	 * @return string
+	 */
+	public function get_format()
+	{
+		return $this->format;
+	}
+
+	/**
+	 * Sets the default string format of the log entry
+	 * 
+	 * @param string $format
+	 * @return Log_Writer
+	 */
+	public function set_format($format)
+	{
+		$this->format = (string) $format;
+		
+		return $this;
+	}
+	
+	/**
+	 * Gets the timezone of this Log_Writer
+	 * 
+	 * @return string
+	 */
+	public function get_timezone()
+	{
+		return $this->timezone;
+	}
+
+	/**
+	 * Sets the timezone of this Log_Writer
+	 * 
+	 * @param string $timezone
+	 * @throws InvalidArgumentException
+	 * @return Log_Writer
+	 */
+	public function set_timezone($timezone)
+	{
+		if ( ! in_array($timezone, DateTimeZone::listIdentifiers()))
+		{
+			throw new InvalidArgumentException(
+				'Argument 1 of set_timezone must be a valid timezone.'
+			);
+		}
+
+		$this->timezone = $timezone;
+
+		return $this;
+	}
+
+	/**
+	 * Gets the log levels that this writer accepts to write
+	 *
+	 * @return array
+	 */
+	public function get_levels()
+	{
+		return $this->levels;
+	}
+
+	/**
+	 * Sets the log levels that this writer accepts to write
+	 *
+	 * @param array $levels
+	 * @throws InvalidArgumentException
+	 * @return Log_Writer
+	 */
+	public function set_levels(array $levels)
```

Comment:
Well, I see that it's a bit confusing. We can add just a note about it instead of sticking with a single set. Otherwise, it may be more confusing that the logger accepts anything, but the writer sticks with a single set?